maven-archetype-quickstart/src/main/resources-filtered/archetype-resources/pom.xml

```diff
@@ -1,11 +1,11 @@
 <?xml version="1.0" encoding="UTF-8"?>
-#macro( compilerProperties $javaCompilerVersion )
+#macro( compilerProperties $javaCompilerVersionLocal )
```

Comment:
Is this variable set by user code/properties somewhere that will need to update to the new name?

Reply:
no.  TL/DR, yes and no,which was the reasoning of the PR.   javaCompilerVersion is a property which can (and should) be used to set desired bytecode version, however in maven-archetype-quickstart it is partially broken (https://issues.apache.org/jira/projects/MARCHETYPES/issues/MARCHETYPES-77) .
In the line of question, however it is not the variable, but it's macro-local copy.  See the calls:
https://github.com/apache/maven-archetypes/blob/bb9279764c5417042daf1eda8f10842a0a5bbf29/maven-archetype-quickstart/src/main/resources-filtered/archetype-resources/pom.xml#L49
and
https://github.com/apache/maven-archetypes/blob/bb9279764c5417042daf1eda8f10842a0a5bbf29/maven-archetype-quickstart/src/main/resources-filtered/archetype-resources/pom.xml#L51

When I was trying to fix the https://issues.apache.org/jira/projects/MARCHETYPES/issues/MARCHETYPES-77, this local occurrence was still meddling  in.

Note, that this PR suggest two names. I prefer the one ion first commit, but the second is probably better. I'm ok with any name you would be happy with. TY!